src/contracts/exchange/NiftyswapExchange20.sol

```diff
@@ -455,6 +455,60 @@ contract NiftyswapExchange20 is ReentrancyGuard, ERC1155MintBurn, INiftyswapExch
     emit LiquidityAdded(_provider, _tokenIds, _tokenAmounts, currencyAmounts);
   }
 
+  /**
+   * @dev Convert pool participation into amounts of token and currency.
+   * @dev Rounding error of the asset with lower resolution is traded for the other asset.
+   * @param _amountPool       Participation to be converted to tokens and currency.
+   * @param _tokenReserve     Amount of tokens on the AMM reserve.
+   * @param _currencyReserve  Amount of currency on the AMM reserve.
+   * @param _totalLiquidity   Total liquidity on the pool.
+   *
+   * @return currencyAmount Currency corresponding to pool amount plus rounded tokens.
+   * @return tokenAmount    Token corresponding to pool amount plus rounded currency.
+   */
+    function _toRoundedLiquidity(
```

Comment:
We probably should apply the royalty logic here as well, otherwise it creates a loopwhole. Only needs to be done in the ERC20-ERC1155 one since the other contract doesn't have that logic.